### ISSUE TITLE
workflow suggestion r.e. individual/many & components

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#90]
how about if on presenting the image, the UI asks you:-

* how many <$x> are there
  [one] [a few] [many]

'many' indicates forest, traffic, crowds etc
'one' is suitable if the whole image focusses on one

-> then in the case of 'one', the UI could go straight onto marking up the components (in label me I liked to do the head/hands/feet/elbows/knees to get the outline.. for cars you can do 'headlights'/'taillights'/'bumper'/'wheels'/'roof')

in the case of 'a few' you could go and draw boxes around each instance;

in the case of 'many', it could say 'annotate the area covered by <$plural form of x>'

having said that.. you might still want a combination of options so even this would leave some situations ambiguous/unsatisfactory (a few clear foreground + many background is a common case)

**Comment by bbernhard** [comment]
> basically with scenes the workflow the tool currently presents is really difficult to use.. I wonder if they break the original intention and design 

agreed. That's one of the (many) "lessons learned". Originally I had only simple images (where the majority of the image is covered by one big object) in mind. But with more complex scenes this approach is obviously not working that well. 

> how about if on presenting the image, the UI asked you:-

>  *  how many <$x> are there
 >   [one] [a few] [many]

Interesting idea! Where would you show the dialog? Should that be shown when you add a new label or right before annotation? At the moment there is a clear distinction between labeling and annotation (with all its advantages and disadvantages). If we would go with the latter approach, we would weaken that separation a bit (as we would now add additional labels to the image depending on <$x>). 

We also need to make clear to the user that the selected option impacts the annotation granularity. I could imagine that sometimes you want to annotate a person in all its details (i.e head, hands, feet), even if there is more than one person in the picture. e.q if the picture focuses on two people talking. If you would go with the "a few" option in that case, you would lose the possibility to do that (if I got you right). 

Another thing is label != annotation. I guess it might be sometimes useful to add a label, although you can't annotate it properly. e.q you might be interested, that there is a "head" in the picture, but you can't annotate it properly as it's too far away. But already the fact that there is a head could be useful for training.

**Comment by dobkeratops** [comment]
>> "But already the fact that there is a head could be useful for training."

absolutely, IMO 'more information is always better'
